torch/testing/_internal/common_methods_invocations.py

```diff
@@ -19158,10 +19166,25 @@ def __init__(
         "_refs.round",
         torch_opinfo_name="round",
     ),
+    ElementwiseUnaryPythonRefInfo(
+        "_refs.sigmoid",
+        torch_opinfo_name="sigmoid",
+        decorators=(
+            # disable complex because of nan, inf
+            DecorateInfo(unittest.skip("Skipped complex tests because of nan, inf inputs"),
```

Comment:
What's going on with nan/inf inputs in complex?

Reply:
The jiterator CUDA sigmoid kernel has different behavior with NaN, INF inputs than the regular aten kernel.
The sigmoid ref matches the regular aten behavior.

Reply:
I see... skipping the entire test on complex inputs doesn't seem ideal when extremal handling is distinct -- we should probably think about adding a mechanism (not in this PR) to specify there's a discrepancy for extremal values only

Reply:
I saw that extremal values were skipped in the main test. Does the reference test also skip them?
https://github.com/pytorch/pytorch/blob/9f5c9617f1666c6611d921a632122b55b8816979/torch/testing/_internal/common_methods_invocations.py#L17158-L17162

Reply:
No, the reference test currently combines everything. We could separate out the extremal part as one option to address the above.

Reply:
I added a flag to skip the extremal values for complex dtypes as a workaround.